react-stopwatch-component/src/stopwatch.jsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+
+class StopWatch extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { time: 0, playing: false };
+    this.handleTimer = null;
+    this.handleClick = this.handleClick.bind(this);
+    this.resetState = this.resetState.bind(this);
+  }
+
+  handleClick() {
+    if (this.state.time > 1 && !this.state.playing) {
+      this.setState({ time: 0, playing: false });
+    } else {
+      this.setState({ time: this.state.time + 1, playing: !this.state.playing });
+    }
+    // this.setState({ time: this.state.time + 1, playing: !this.state.playing });
+
+  }
+
+  handleCounter() {
+    this.handleTimer = setInterval(() => {
+      this.setState({ time: this.state.time + 1 });
+    }, 1000);
+  }
+
+  resetState() {
+    this.setState({ time: 0 });
+    this.handleClick();
+  }
+
+  render() {
+    const { time, playing } = this.state;
+
+    if (time === 1) { this.handleCounter(); } else if (!playing) { clearInterval(this.handleTimer); }
```

Comment:
It would be better to put this logic in your `handleClick`. You want to avoid calling methods that update state directly in `render` in this case this can lead to multiple intervals running or clearInterval being called multiple times unnecessarily. This might work as is but it is prone to bugs.